File: fila.py
class Fila:
    def __init__(self):
        self.elementos = []

    # Inserir elemento na fila
    def enqueue(self, item):
        self.elementos.append(item)

    # Remover elemento da frente
    def dequeue(self):
        if not self.is_empty():
            return self.elementos.pop(0)
        return "Fila vazia!"

    # Consultar o primeiro elemento
    def front(self):
        if not self.is_empty():
            return self.elementos[0]
        return "Fila vazia!"

    # Verificar se está vazia
    def is_empty(self):
        return len(self.elementos) == 0

    # Mostrar fila
    def mostrar_fila(self):
        print("Fila:", self.elementos)


if __name__ == "__main__":
    fila = Fila()

    fila.enqueue("Luis")
    fila.enqueue("Manuel")
    fila.enqueue("Marques")
    fila.enqueue("Adilson")
    fila.enqueue("Emiliano")

    fila.mostrar_fila()

    print("Removido:", fila.dequeue())
    print("Em frente:", fila.front())
    print("Removido:", fila.dequeue())
    print("Em frente:", fila.front())
    print("Está vazia?", fila.is_empty())

    fila.mostrar_fila()

File: Fila.java
import java.util.ArrayList;

public class Fila {
    private ArrayList<String> elementos;

    // Construtor
    public Fila() {
        elementos = new ArrayList<>();
    }

    // Inserir elemento na fila
    public void enqueue(String item) {
        elementos.add(item);
    }

    // Remover elemento da frente da fila
    public String dequeue() {
        if (!isEmpty()) {
            return elementos.remove(0);
        }
        return "Fila vazia!";
    }

    // Consultar o primeiro elemento
    public String front() {
        if (!isEmpty()) {
            return elementos.get(0);
        }
        return "Fila vazia!";
    }

    // Verificar se a fila está vazia
    public boolean isEmpty() {
        return elementos.isEmpty();
    }

    // Mostrar elementos da fila
    public void mostrarFila() {
        System.out.println("Fila: " + elementos);
    }

    public static void main(String[] args) {
        Fila fl = new Fila();

        fl.enqueue("Luis");
        fl.enqueue("Manuel");
        fl.enqueue("Marques");
        fl.enqueue("Adilson");
        fl.enqueue("Emiliano");

        fl.mostrarFila();

        System.out.println("Removido: " + fl.dequeue());
        System.out.println("Em frente: " + fl.front());
        System.out.println("Removido: " + fl.dequeue());
        System.out.println("Em frente: " + fl.front());
        System.out.println("Está vazia? " + fl.isEmpty());

        fl.mostrarFila();
    }
}